From 347aa29bbd516be9b498b5910ab6b4f905235981 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Mon, 18 Mar 2019 10:06:15 -0700
Subject: [PATCH] TRITON-884 remove cloud analytics (ca) from Triton Reviewed
 by: Tim Foster <tim.foster@joyent.com> Approved by: Tim Foster
 <tim.foster@joyent.com>

---
 Makefile        | 59 -------------------------------------------------
 targets.json.in | 32 ---------------------------
 2 files changed, 91 deletions(-)

diff --git a/Makefile b/Makefile
index ff97a59..6ccfb43 100644
--- a/Makefile
+++ b/Makefile
@@ -171,65 +171,6 @@ clean_amon:
 	$(RM) -rf $(BITS_DIR)/amon
 	(cd build/sdc-amon && gmake clean)
 
-#---- cloud-analytics
-#TODO:
-# - merge CA_VERSION and CA_PUBLISH_VERSION? what about the version sed'd into
-#   the package.json's?
-# - look at https://hub.joyent.com/wiki/display/dev/Setting+up+Cloud+Analytics+development+on+COAL-147
-#   for env setup. Might be demons in there. (RELENG-192)
-
-_ca_stamp=$(SDC_CLOUD_ANALYTICS_BRANCH)-$(TIMESTAMP)-g$(SDC_CLOUD_ANALYTICS_SHA)
-CA_BITS=$(BITS_DIR)/ca/ca-pkg-$(_ca_stamp).tar.gz \
-	$(BITS_DIR)/ca/cabase-$(_ca_stamp).tar.gz \
-	$(BITS_DIR)/ca/cainstsvc-$(_ca_stamp).tar.gz
-CA_BITS_0=$(shell echo $(CA_BITS) | awk '{print $$1}')
-CA_IMAGE_BIT=$(BITS_DIR)/ca/ca-zfs-$(_ca_stamp).zfs.gz
-CA_MANIFEST_BIT=$(BITS_DIR)/ca/ca-zfs-$(_ca_stamp).imgmanifest
-CA_BASE_BIT=$(BITS_DIR)/ca/cabase-$(_ca_stamp).tar.gz
-CA_BASE_MANIFEST_BIT=$(BITS_DIR)/ca/cabase-$(_ca_stamp).manifest
-CA_INSTSVC_BIT=$(BITS_DIR)/ca/cainstsvc-$(_ca_stamp).tar.gz
-CA_INSTSVC_MANIFEST_BIT=$(BITS_DIR)/ca/cainstsvc-$(_ca_stamp).manifest
-
-.PHONY: ca
-ca: $(CA_BITS_0) ca_image
-
-# PATH for ca build: Ensure /opt/local/bin is first to put gcc 4.5 (from
-# pkgsrc) before other GCCs.
-$(CA_BITS): build/sdc-cloud-analytics
-	@echo "# Build ca: branch $(SDC_CLOUD_ANALYTICS_BRANCH), sha $(SDC_CLOUD_ANALYTICS_SHA), time `date -u +%Y%m%dT%H%M%SZ`"
-	mkdir -p $(BITS_DIR)
-	(cd build/sdc-cloud-analytics && NPM_CONFIG_CACHE=$(MG_CACHE_DIR)/npm TIMESTAMP=$(TIMESTAMP) BITS_DIR=$(BITS_DIR) PATH="/sbin:/opt/local/bin:/usr/gnu/bin:/usr/bin:/usr/sbin:$(PATH)" gmake clean pkg release publish)
-	@echo "# Created ca bits (time `date -u +%Y%m%dT%H%M%SZ`):"
-	@ls -l $(CA_BITS)
-	@echo ""
-
-.PHONY: ca_image
-ca_image: $(CA_IMAGE_BIT)
-
-$(CA_IMAGE_BIT): $(CA_BITS_0)
-	@echo "# Build ca_image: branch $(SDC_CLOUD_ANALYTICS_BRANCH), sha $(SDC_CLOUD_ANALYTICS_SHA), time `date -u +%Y%m%dT%H%M%SZ`"
-	./tools/prep_dataset_in_jpc.sh -i "$(CA_IMAGE_UUID)" -t $(CA_BITS_0) \
-		-o "$(CA_IMAGE_BIT)" -p $(CA_PKGSRC) -O "$(MG_OUT_PATH)" \
-		-t $(CA_EXTRA_TARBALLS) -n $(CA_IMAGE_NAME) \
-		-v $(_ca_stamp) -d $(CA_IMAGE_DESCRIPTION)
-	@echo "# Created ca image (time `date -u +%Y%m%dT%H%M%SZ`):"
-	@ls -l $$(dirname $(CA_IMAGE_BIT))
-	@echo ""
-
-ca_publish_image: $(CA_IMAGE_BIT)
-	@echo "# Publish ca images to SDC Updates repo."
-	$(UPDATES_IMGADM) import -ddd -m $(CA_MANIFEST_BIT) -f $(CA_IMAGE_BIT)
-	$(UPDATES_IMGADM) import -ddd -m $(CA_BASE_MANIFEST_BIT) -f $(CA_BASE_BIT)
-	$(UPDATES_IMGADM) import -ddd -m $(CA_INSTSVC_MANIFEST_BIT) -f $(CA_INSTSVC_BIT)
-
-# Warning: if CA's submodule deps change, this 'clean_ca' is insufficient. It would
-# then need to call 'gmake dist-clean'.
-clean_ca:
-	$(RM) -rf $(BITS_DIR)/ca
-	(cd build/sdc-cloud-analytics && gmake clean)
-
-
-
 #---- UFDS
 
 
diff --git a/targets.json.in b/targets.json.in
index ebb4306..388e73b 100644
--- a/targets.json.in
+++ b/targets.json.in
@@ -38,7 +38,6 @@ function build_headnode_target()
       "assets",
       "registrar",
       "binder",
-      "ca",
       "cloudapi",
       "cnapi",
       "dhcpd",
@@ -100,36 +99,6 @@ cat <<EOF
     "public": true,
     "deps": []
   },
-  "ca": {
-    "appliance": "true",
-    "build_platform": "20151126T062538Z",
-    "image_name": "ca",
-    "image_description": "SDC Cloud Analytics",
-    "image_version": "1.0.0",
-    "image_uuid": "fd2cc906-8938-11e3-beab-4359c665ac99",
-    "pkgsrc": [
-      "zlib-1.2.3",
-      "png-1.5.9",
-      "openssl-0.9.8w",
-      "GeoIP-1.4.8",
-      "GeoLiteCity-201203"
-    ],
-    "repos": [
-      {"url": "$MG_GIT_PREFIX/sdc-cloud-analytics.git", "submodule-update": false}
-    ],
-    "public": true,
-    "deps": [
-      "amon",
-      "config-agent",
-      "registrar",
-      "https://download.joyent.com/pub/build/sdcnode"
-    ],
-    "tarballs": [
-      {"tarball": "registrar/registrar-pkg-*.tar.gz", "sysroot": "/"},
-      {"tarball": "amon/amon-agent-*.tgz", "sysroot": "/opt"},
-      {"tarball": "config-agent/config-agent-*.tar.gz", "sysroot": "/opt/smartdc"}
-    ]
-  },
   "amon": {
     "appliance": "true",
     "build_platform": "20151126T062538Z",
@@ -1352,7 +1321,6 @@ cat <<EOF
       "net-agent",
       "vm-agent",
       "amon",
-      "ca",
       "smartlogin",
       "marlin",
       "firewaller",
-- 
2.21.0

